src/Extensions/ElementalAreasExtension.php

```diff
@@ -340,4 +344,23 @@ public function requireDefaultRecords()
 
         $this->owner->extend('onAfterRequireDefaultElementalRecords');
     }
+
+    /**
+     * Whether it's OK to alter the ElementalArea based the current versioned reading mode
+     */
+    private function canAlterElementalArea(): bool
+    {
+        $readingMode = Versioned::get_reading_mode();
+        // Reading mode is Stage.Stage, which is the normal state on the CMS
+        if ($readingMode === 'Stage.' . Versioned::DRAFT) {
+            return true;
+        }
+        // When running unit tests he Versioned reading mode is empty string by default as a result
+        // of Versioned::reset() which is called because of the initial flush
+        // Treat this the same as Stage.Stage
+        if ($readingMode === '' && Director::is_cli()) {
+            return true;
+        }
```

Comment:
CLI can also be `sake db:build` or `sake tasks:some-specific-task` - but in those cases the reading mode should be set. If the reading mode isn't set in those scenarios, we probably either have a bug in the way sake bootstraps versioning, or someone is doing something weird.

I assume this is specifically for tests. We should avoid having weird stuff like this in tests. Instead, just set a valid reading mode in the test.

Reply:
Yup will be Stage.Stage for a BuildTask

Will update tests

Reply:
I'll go find out why this is happening incorrectly in behat. Might be a bit, it's not my top priority.
We should not have weird stuff like this (which could affect actual projects running in live mode) in source code just to support tests.

Reply:
Behat calls `reset()` on all classes that implement the `Resettable` interface in `SilverStripeContext::before()`.
In `Versioned::reset()` this sets `reading_mode` to an empty string.

This means that for the CLI context (which is used for creating fixture records), the reading mode is empty and the stage is empty.

Adding this to the bottom of that method seems to work:
```php
if (class_exists(Versioned::class)) {
    Versioned::set_stage(Versioned::DRAFT);
}
```

Reply:
Where is that call to `Versioned::reset()` happening?

Reply:
My comment about lists a pseudo stack trace

Reply:
Oh I didn't realise that was meant to be a stack trace, thanks for clearing that up. I'll go see why that's interfering with setting the stage in the separate PRs.

I would normally just ask you to do it since I'm just meant to be reviewing but I feel like you're gonna be stubborn about this :sweat_smile: